libfs/fs.go

```diff
@@ -108,31 +109,60 @@ func (e ErrNotADirectory) Error() string {
 	return fmt.Sprintf("%s is not a directory", e.Name)
 }
 
+type accessType int
+
+const (
+	readwrite accessType = iota
+	readwriteNoCreate
+	readonly
+)
+
 func newFS(ctx context.Context, config libkbfs.Config,
 	tlfHandle *libkbfs.TlfHandle, branch libkbfs.BranchName, subdir string,
-	uniqID string, priority keybase1.MDPriority, unwrap, readonly bool) (
-	*FS, error) {
+	uniqID string, priority keybase1.MDPriority, unwrap bool,
+	atype accessType) (*FS, error) {
 	rootNodeGetter := config.KBFSOps().GetOrCreateRootNode
-	if branch != libkbfs.MasterBranch {
+	if branch != libkbfs.MasterBranch || atype != readwrite {
 		rootNodeGetter = config.KBFSOps().GetRootNode
 	}
 
 	rootNode, ei, err := rootNodeGetter(ctx, tlfHandle, branch)
 	if err != nil {
 		return nil, err
 	}
+	log := config.MakeLogger("")
+	if rootNode == nil {
+		if len(subdir) > 0 {
+			return nil, errors.Errorf(
+				"Subdir %s doesn't exist in empty TLF", subdir)
+		}
+		log.CDebugf(ctx,
+			"Returning empty FS for empty TLF=%s", tlfHandle.GetCanonicalName())
+		return &FS{
+			ctx: ctx,
+			fsInner: &fsInner{
```

Comment:
should this behave differently in the event that it's being opened read-only?

Reply:
I don't think so.  An empty file system is inherently read-only already, since it doesn't allow any write operations.